l10n_ch_payment_slip/report/ir_action.py

```diff
@@ -1,46 +1,172 @@
+
+        # Manual cleanup of the temporary files
+        for temporary_file in temporary_files:
+            try:
+                os.unlink(temporary_file)
+            except (OSError, IOError):
+                _logger.error(
+                    'Error when trying to remove file %s' % temporary_file)
+
+        return content
+
+    @api.model
+    def _get_report_from_name(self, report_name):
+        """Return also report of report_type reportlab-pdf and not only qweb
         """
-        context = self._context
-        if context is None:
-            context = {}
-        if context.get('force_pdf'):
-            new_report = self._lookup_report(name)
-            if isinstance(new_report, (str, unicode)):
-                return self.env['report'].get_pdf(
-                    res_ids,
-                    new_report,
-                    data=data,
-                    ), 'pdf'
-        return super(IrActionsReportXMLReportlab, self).render_report(
-            res_ids,
-            name,
-            data,
+        report = super()._get_report_from_name(report_name)
+        if not report:
+            report = self.search([
+                ('report_type', '=', 'reportlab-pdf'),
+                ('report_name', '=', report_name)],
+                limit=1
+            )
+        return report
+
+    @api.multi
+    def render_reportlab_pdf(self, res_ids=None, data=None):
+        if (self.report_name != 'l10n_ch_payment_slip.'
+                'one_slip_per_page_from_invoice') or not res_ids:
+            return
+
+        save_in_attachment = {}
+
+        # Dispatch the records by ones having an attachment and ones
+        # requesting a call to reportlab. (copied from render_qweb_pdf)
+        Model = self.env[self.model]
+        records = Model.browse(res_ids)
+        rl_records = Model
+        if self.attachment:
+            for rec in records:
+                attachment_id = self.retrieve_attachment(rec)
+                if attachment_id:
+                    save_in_attachment[rec.id] = attachment_id
+                if not self.attachment_use or not attachment_id:
+                    rl_records += rec
+        else:
+            rl_records = records
+        res_ids = rl_records.ids
+
+        if save_in_attachment and not res_ids:
+            _logger.info('The PDF report has been generated from attachments.')
+            return self._post_pdf(save_in_attachment), 'pdf'
+
+        pdf_content = self._generate_one_slip_per_page_from_invoice_pdf(
+            res_ids)
+        if res_ids:
+            _logger.info(
+                'The PDF report has been generated for records %s.' % (
+                    str(res_ids))
             )
+            return self._post_pdf(save_in_attachment, pdf_content=pdf_content,
+                                  res_ids=res_ids), 'pdf'
+        return pdf_content, 'pdf'
+
+    # TODO consider https://github.com/OCA/reporting-engine/issues/241
```

Comment:
Time to do this TODO ? [https://github.com/OCA/reporting-engine/issues/241](https://github.com/OCA/reporting-engine/issues/241)

Reply:
This payment slip is about to die, the new format using QR code will be used in few months.

I think we can drop the TODO.